code-snippet-plugin/src/main/java/com/okta/maven/snippet/CodeSnippetMojo.java

```diff
@@ -115,15 +118,30 @@ public void execute() throws MojoExecutionException {
 
         final Map<String, String> methodBodyMap = new HashMap<>();
 
-
         new VoidVisitorAdapter<Object>() {
             @Override
             public void visit(final MethodDeclaration method, final Object arg) {
                 super.visit(method, arg);
 
                 String body = method.getBody().get().getStatements().stream()
-                        .map(Node::toString)
+                        .map(statement -> {
+                            TokenRange tokenRange = statement.getTokenRange().get();
+                            Optional<Comment> comment = statement.getComment();
+
+                            StringBuilder blockAString = new StringBuilder();
+                            if ( comment.isPresent()) {
+                                blockAString.append("\n")
+                                            .append(comment.get());
+                            }
+                            blockAString.append(tokenRange.toString().replaceAll(" {8}", ""));
+
+                            return blockAString;
+                        })
                         .collect(Collectors.joining("\n"));
```

Comment:
Here, it appears we are joining together this String based on newlines. If the `comment` contains a block string (with newlines needed) - won't this malform our expected output?

As an alternative, could you use a special character sequence `<-->`, or something of the sort as your delimiter?

Reply:
This joins multiple statements together. given the two statements:
```
int one = 1;
```
and
```
int two = 2;
```
The output would match what is in the source file:

```
int one = 1;
int two = 2;
```